File: languages/django_models/zone.py
from .abstract_element_model import AbstractElementModel
from django.db import models

class Zone(AbstractElementModel):

    # Scope
    role = models.TextField(blank=True, null=True)
    start_date = models.PositiveIntegerField(blank=True, null=True)
    end_date = models.PositiveIntegerField(blank=True, null=True)
    phenomena = models.ManyToManyField("Phenomenon", blank=True, related_name="zone_phenomena")
    linked_zones = models.ManyToManyField("Zone", blank=True, related_name="zone_linked_zones")

    # World
    context = models.TextField(blank=True, null=True)
    populations = models.ManyToManyField("Collective", blank=True, related_name="zone_populations")
    titles = models.ManyToManyField("Title", blank=True, related_name="zone_titles")
    principles = models.ManyToManyField("Construct", blank=True, related_name="zone_principles")

    def __str__(self):
        return self.name

File: languages/csharp_unity_json_properties/Zone.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;
using Utils;

[System.Serializable]
public class Zone : Element
{
    [JsonProperty("role"), TextAttribute("")]
    public string role;
    [JsonProperty("start_date"), Integer(0)]
    public int startDate;
    [JsonProperty("end_date"), Integer(0)]
    public int endDate;
    [JsonProperty("phenomena"), ReferenceAttribute(typeof(Phenomenon), true)]
    public string phenomena;
    [JsonProperty("linked_zones"), ReferenceAttribute(typeof(Zone), true)]
    public string linkedZones;
    [JsonProperty("context"), TextAttribute("")]
    public string context;
    [JsonProperty("populations"), ReferenceAttribute(typeof(Collective), true)]
    public string populations;
    [JsonProperty("titles"), ReferenceAttribute(typeof(Title), true)]
    public string titles;
    [JsonProperty("principles"), ReferenceAttribute(typeof(Construct), true)]
    public string principles;
}
